test/url-filters.test.ts

```
@@ -0,0 +1,74 @@
+  })
+})
+
+describe('url-filters: category slug validation', () => {
+  it('accepts legacy category slugs and tots by default', () => {
+    expect(isValidCategorySlug('concerts')).toBe(true)
+    expect(isValidCategorySlug('tots')).toBe(true)
+  })
+
+  it('accepts dynamic categories when provided', () => {
+    const dynamic = [{ id: 1, name: 'Fires', slug: 'fires-i-mercats' }]
+    expect(isValidCategorySlug('fires-i-mercats', dynamic as any)).toBe(true)
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)